packages/fast-development-site-react/src/components/site/dev-tools-code-preview.tsx

```diff
@@ -0,0 +1,298 @@
+import * as React from "react";
+import { frameworkEnum } from "./dev-tools";
+import { IFormChildOption } from "./";
+import { get, isEmpty, isObject, set, uniqueId } from "lodash-es";
+
+export interface ICodePreviewProps {
+    componentName: string;
+    childOptions: IFormChildOption[];
+    framework: frameworkEnum;
+    data: any;
+}
+
+export interface IReactChildren {
+    id: string;
+    propertyName: string;
+    location: string;
+    data: any;
+    component?: any;
+}
+
+export default class CodePreview extends React.Component<ICodePreviewProps, {}> {
+
+    private variables: string;
+
+    public render(): JSX.Element {
+        return (
+            <pre>
+                <code>
+                    {this.getCodePreview()}
+                </code>
+            </pre>
+        );
+    }
+
+    private getCodePreview(): string {
+        let codePreview: string = "TBD";
+        this.variables = "";
+
+        switch (this.props.framework) {
+            case frameworkEnum.react:
+                codePreview = this.generateReactCodePreview(this.props.componentName, this.props.data, "", "");
+                break;
+        }
+
```

Comment:
isn't `this.variables` an empty string here? - is it necessary to concat these?

Reply:
no, this.variables gets added to during the this.generateReactCodePreview()